bar.go

```diff
@@ -22,23 +23,23 @@ var (
 type barStackListener struct {
 }
 
-func barCreateIcon(taskbar bool, idata *IconData, win Window) *barIcon {
-	if idata == nil || idata.IconPath == "" {
+func barCreateIcon(taskbar bool, idata driver.IconData, win Window) *barIcon {
```

Comment:
Having the top level code call into the driver is a backwards dependency - I think the IconData type is in the wrong package.